pkgs/development/python-modules/aws-adfs/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, buildPythonPackage, fetchPypi, pythonPackages, isPy3k
+, pytest, pytestrunner, pytestcov, mock, lxml, boto3, requests, click, configparser }:
+
+buildPythonPackage rec {
+  version = "0.12.0";
+  pname = "aws-adfs";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1cjrm61k6905dmhgrqyc5caxx5hbhj3sr6cx4r6sbdyz453i7pc6";
+  };
+
+  # Relax version constraint
+  patchPhase = ''
+    sed -i 's/coverage < 4/coverage/' setup.py
+  '';
+
+  # Test suite writes files to $HOME/.aws/, or /homeless-shelter if unset
+  HOME = ".";
+
+  checkInputs = [ pytest pytestrunner pytestcov mock ];
+  propagatedBuildInputs = [ lxml boto3 requests click configparser ];
+
+  # Python3 tests fail with UTF-8 issues due to click
+  doCheck = !isPy3k;
```

Comment:
Click also has quite a few python3 UTF-8 errors in its own test suite, but it happens to be disabled at the moment.  It looks like they're preparing a click 7.0 which might fix a handful of locale issues, but 6.7 is still the latest one available as of now.

Reply:
You may have luck with adding `glibcLocales` to `checkInputs` and setting `LC_ALL = "en_US.UTF-8"`.

Reply:
Aha, I was setting `LC_ALL` but passing `locale` to `checkInputs`; thanks for the tip!  It works now.